src/test/java/com/aws/greengrass/clientdevices/auth/certificate/BackgroundCertificateRefreshTest.java

```diff
@@ -190,4 +222,184 @@ void GIVEN_certsAndThings_WHEN_certificateAssociatedToMoreThanOneThing_THEN_itDo
         assertEquals(pemStore.getPem(certA.getCertificateId()).get(), certificateAPem);
     }
 
+    @Test
+    void GIVEN_networkWasDown_WHEN_networkUp_THEN_backgroundTaskTriggered() {
+        backgroundRefresh.accept(NetworkState.ConnectionState.NETWORK_DOWN);
+        verify(backgroundRefresh, times(0)).run();
+        backgroundRefresh.accept(NetworkState.ConnectionState.NETWORK_UP);
+        verify(backgroundRefresh, times(1)).run();
+        backgroundRefresh.accept(NetworkState.ConnectionState.NETWORK_DOWN);
+        verify(backgroundRefresh, times(1)).run();
+    }
+
+    @Test
+    void GIVEN_deviceOnline_WHEN_iotCoreCallFails_THEN_itShouldRetryAgainWhenCalled() {
+        AccessDeniedException accessDeniedException = AccessDeniedException.builder().build();
+        when(iotAuthClientFake.getThingsAssociatedWithCoreDevice()).thenThrow(accessDeniedException);
+
+        backgroundRefresh.run();
+        verify(iotAuthClientFake, times(1)).getThingsAssociatedWithCoreDevice();
+
+        backgroundRefresh.run();
+        verify(iotAuthClientFake, times(2)).getThingsAssociatedWithCoreDevice();
+    }
+
+    @Test
+    void GIVEN_networkChangesAndScheduledByTime_WHEN_triggered_THEN_shouldNotRunMoreThanOnceADay() {
+        Instant now = Instant.now();
+        mockInstant(now.toEpochMilli());
+
+        verify(iotAuthClientFake, times(0)).getThingsAssociatedWithCoreDevice();
+        backgroundRefresh.run();
+        backgroundRefresh.run();
+        backgroundRefresh.run();
+        verify(iotAuthClientFake, times(1)).getThingsAssociatedWithCoreDevice();
+
+        Instant twentyFiveHoursLater = Instant.now().plus(Duration.ofHours(25));
+        mockInstant(twentyFiveHoursLater.toEpochMilli());
+
+        backgroundRefresh.run();
+        verify(iotAuthClientFake, times(2)).getThingsAssociatedWithCoreDevice();
+    }
+
+    @Test
+    void GIVEN_triggeredByMultipleThreads_WHEN_called_THEN_itShouldOnlyRunOnceForThatTimeWindow()
+            throws InterruptedException {
+       Thread t1 = new Thread(() -> { backgroundRefresh.run(); });
+       Thread t2 = new Thread(() -> { backgroundRefresh.run(); });
+       t1.start();
+       t2.start();
+       t1.join();
+       t2.join();
+
+        verify(iotAuthClientFake, times(1)).getThingsAssociatedWithCoreDevice();
```

Comment:
this isn't necessarily guaranteed, there would still be a (small) possibility that t1 completes before t2 starts.   it's probably ok in practice, just wanted to call it out